### ISSUE TITLE
tri_gamma is not dispatched with jax

### ISSUE THREAD

**Issue by ferrine** [pymc-devs/pytensor#497]
### Description

tmp workaround I use
```python
import pytensor
import jax.scipy
from functools import partial
pytensor.scalar.math.TriGamma.nfunc_spec = ("scipy.tri_gamma", 1, 1)
jax.scipy.tri_gamma = partial(jax.scipy.special.polygamma, 1)
```

**Comment by ferrine** [comment]
this is something that is needed btw, but with a more proper implementation


```python

from pytensor.scalar.basic import ScalarOp, upgrade_to_float
from pytensor.tensor.elemwise import Elemwise
from pytensor.gradient import grad_not_implemented

class PolyGamma(ScalarOp):
    nfunc_spec = ("scipy.special.polygamma", 2, 1)

    @staticmethod
    def st_impl(i, x):
        return scipy.special.polygamma(i, x)

    def impl(self, i, x):
        return self.st_impl(i, x)

    def L_op(self, inputs, outputs, grads):
        (i, x,) = inputs
        (gz,) = grads
        return [grad_not_implemented(self, 0, i), gz * self(i+1, x)]

polygamma = Elemwise(PolyGamma(upgrade_to_float, "polygamma"))
```

**Comment by ricardoV94** [comment]
You don't need to implement the Op just to dispatch to JAX, just implement a custom dispatch?

**Comment by ferrine** [comment]
I want the proper way, the Op does the job in plain sampling

**Comment by ricardoV94** [comment]
Are you requesting a generalized polyagamma Op or just to have the current trigamma properly dispatched in jax?

The latter doesn't demmand a new Op, just the correct dispatch. The default dispatch based on name or `func_spec` is just a convenience to get most things working out of the box but it wasn't expected to work for all cases.

**Comment by ricardoV94** [comment]
A proper dispatch for an Ops with a distinct name looks like this

```python
@jax_funcify.register(TriGamma)
def jax_funcify_Triga.ma(op, **kwargs):

    def trigamma(x)v      
        return jax.scipy.special.polyagamma(1, x)

    return trigamma
```

**Comment by ferrine** [comment]
The below snippet did not work due to all the arguments not being scalars. Generic polygamma is also needed implemented properly since you anyway will end up with taking gradients which's are easier to implement with polygamma

**Comment by ricardoV94** [comment]
Why would all arguments need to be scalars? What error are you getting exactly?

The dispatch of Elemwise should just be calling vectorize on the dispatch of Scalar. Can you share the full snippet you're running?
